lib/Workflow/Notification/NotificationEmailService.php

```diff
@@ -78,7 +78,19 @@ public function sendWorkflowEmailNotification(array $users, array $roles, Workfl
             $deeplink = '';
             $hostUrl = Tool::getHostUrl();
             if ($hostUrl !== '') {
-                $deeplink = $hostUrl . $this->router->generate('pimcore_admin_login') . '/deeplink?object_' . $subject->getId() . '_object';
+                
+                // Decide what kind of link to create
+                $objectType = $type = "object";
+                if ($subject instanceof \Pimcore\Model\Document) {
+                    $objectType = "document";
+                    $type = $subject->getType();
+                }
+                if ($subject instanceof \Pimcore\Model\Asset) {
+                    $objectType = "asset";
+                    $type = $subject->getType();
+                }
+
+                $deeplink = $hostUrl . $this->router->generate('pimcore_admin_login') . '/deeplink?'.$objectType.'_' . $subject->getId() . '_'. $type;
```

Comment:
We should also use the correct route:
```suggestion
                $deeplink = $hostUrl . $this->router->generate('pimcore_admin_login_deeplink') . '?'.$objectType.'_' . $subject->getId() . '_'. $type;
```